tests/unit/LatteSwapOptimalDeposit.test.ts

```diff
@@ -0,0 +1,295 @@
+          token1Amount,
+          0,
+          aliceAddress,
+          MaxUint256
+        )
+      )
+        .to.emit(sortedToken0, "Transfer")
+        .withArgs(deployerAddress, latteSwapOptimalDeposit.address, token0Amount)
+        .to.emit(sortedToken1, "Transfer")
+        .withArgs(deployerAddress, latteSwapOptimalDeposit.address, token1Amount)
+        .to.emit(sortedToken0, "Transfer")
+        .withArgs(latteSwapOptimalDeposit.address, nonNativePair.address, token0Amount)
+        .to.emit(sortedToken1, "Transfer")
+        .withArgs(latteSwapOptimalDeposit.address, nonNativePair.address, token1Amount)
+        .to.emit(nonNativePair, "Transfer")
+        .withArgs(AddressZero, AddressZero, MINIMUM_LIQUIDITY)
+        .to.emit(nonNativePair, "Transfer")
+        .withArgs(AddressZero, aliceAddress, expectedLiquidity.sub(MINIMUM_LIQUIDITY))
+        .to.emit(nonNativePair, "Sync")
+        .withArgs(token0Amount, token1Amount)
+        .to.emit(nonNativePair, "Mint")
+        .withArgs(latteSwapRouter.address, token0Amount, token1Amount);
+
+      expect(await nonNativePair.balanceOf(aliceAddress)).to.eq(expectedLiquidity.sub(MINIMUM_LIQUIDITY));
+    });
+
+    it("should add an additional liquidity", async () => {
+      const [sortedToken0, sortedToken1] = sortsBefore(token0.address, token1.address)
+        ? [token0, token1]
+        : [token1, token0];
+      const token0Amount = ethers.utils.parseEther("10000");
+      const token1Amount = ethers.utils.parseEther("10000");
+      const aliceAddress = await alice.getAddress();
+      const deployerAddress = await deployer.getAddress();
+      // sqrt(1*1) - MINIMUM_LIQUIDITY
+      // 1*10^18 - 1000
+      const expectedLiquidity = ethers.utils.parseEther("10000");
+      // approve tokens
+      await sortedToken0.approve(latteSwapOptimalDeposit.address, MaxUint256);
+      await sortedToken1.approve(latteSwapOptimalDeposit.address, MaxUint256);
+
+      // add initial liquidity
+      await latteSwapOptimalDeposit.optimalAddLiquidity(
+        sortedToken0.address,
+        sortedToken1.address,
+        token0Amount,
+        token1Amount,
+        0,
+        deployerAddress,
+        MaxUint256
+      );
+
+      // add additional liquidity
+      const token0Amount2 = ethers.utils.parseEther("4");
+      const token1Amount2 = ethers.utils.parseEther("2");
+      await expect(
+        latteSwapOptimalDeposit.optimalAddLiquidity(
+          sortedToken0.address,
+          sortedToken1.address,
+          token0Amount2,
+          token1Amount2,
+          0,
+          aliceAddress,
+          MaxUint256
+        )
+      )
+        .to.emit(sortedToken0, "Transfer")
+        .withArgs(deployerAddress, latteSwapOptimalDeposit.address, token0Amount2)
+        .to.emit(sortedToken1, "Transfer")
+        .withArgs(deployerAddress, latteSwapOptimalDeposit.address, token1Amount2);
+
+      expect(await nonNativePair.balanceOf(aliceAddress)).to.eq("2998698513177453476");
```

Comment:
can you comment the details for deriving 2998698513177453476?, 
- eg, how much to swap x to y, and for addding x and y having the same value into a liquidity, what's the calculation based on existing LP (supply)